src/main/geometries/GeometryList.java

```diff
@@ -30,12 +33,19 @@ public GeometryList(Geometry... geometries) {
 		add(geometries);
 	}
 
+	private GeometryList(BoundingBox border, Intersectible... intersectibles) {
+		for (Intersectible intersectible : intersectibles) {
+			this.intersectibles.add(intersectible);
+		}
+		this.border = border; // #efficient
+	}
+
 	/**
 	 * Add a single {@link Geometry} to the collection.
 	 *
 	 * @param geometry The {@link Geometry} to add.
 	 */
-	public void add(Geometry geometry) {
+	public void add(Intersectible geometry) {
```

Comment:
```suggestion
	public void add(Geometry geometry) {
```
We dont want the public adding a GeometryList without us flattening it. We only should accept Geometry.

If you need one to accept Intersectible, make another private one that takes an Intersectible with the same implementation, then change this implementation to simply call that function.

Reply:
i already changed it to private